Feat/2664 add argillatraining module for autotrain

davidberenstein1957 opened the pull request:
# Description

added support for AutoTrain via direct Argilla dataset integrations.

Closes #2664

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)


- [X] New feature (non-breaking change which adds functionality)
- [X] Refactor (change restructuring the codebase without changing functionality)
- [X] Improvement (change adding some improvement to an existing functionality)


**How Has This Been Tested**

pytest tests/training/test_autotrain.py

**Checklist**

- [ ] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

davidberenstein1957 committed:
chore: initial outline autotrain integration

davidberenstein1957 committed:
chore: initial working version AutoTrain for textclassification

davidberenstein1957 committed:
chore: added tests to AutoTrain trainer

davidberenstein1957 committed:
fix: resolved failing tests w.r.t. missing env var

frascuchon committed:
Update develop version

alvarobartt committed:
feat: relax `codecov/patch` threshold to avoid failures (#2989)

# Description

Relax the `codecov/patch` threshold to avoid CI from failing too often
due to small code coverage percentage changes.

Closes #2985

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+from argilla.utils.dependency import require_version
+
+
+class AutoTrainer(object):
```
```suggestion
class AutoTrainMixIn:
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+
+
+class AutoTrainer(object):
+    def prepare_dataset(self, data_dict: dict = {}):
```
```suggestion
    def prepare_dataset(self, data_dict_params: Optional[dict] = {}) -> None:
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+        Args:
+          data_dict (dict): `data_dict` is a dictionary that contains additional parameters to be passed
+        to the `AutoTrainDataset` constructor. These parameters are optional and can be used to
+        customize the dataset preparation process.
```
```suggestion
            to the `AutoTrainDataset` constructor. These parameters are optional and can be used to
            customize the dataset preparation process.
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+        mappings.
+
+        Args:
+          data_dict (dict): `data_dict` is a dictionary that contains additional parameters to be passed
```
```suggestion
          data_dict_params: A dictionary that contains additional parameters to be passed
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+            token=self.HF_TOKEN,
+            username=self.AUTOTRAIN_USERNAME,
+            project_name=self.project_name,
+            **data_dict,
```
```suggestion
            **data_dict_params,
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+            model_choice = "HuggingFace Hub"
+        if model_choice == "autotrain":
+            if len(job_params) > 1:
+                raise ValueError("❌ Only one job parameter is allowed for AutoTrain.")
```
Maybe we're more safe if we don't use emojis in the exceptions, so I'd reformat this as:

```suggestion
                raise ValueError("Only one job parameter is allowed for AutoTrain.")
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+        self.project_name = f"{self._workspace}_{self._name}_{str(uuid4())[:8]}"
+
+        if self._seed:
+            self._logger.warning("❌ Setting a seed is not supported by `autotrain-advanced`.")
```
Same here with the emoji:

```suggestion
            self._logger.warning("Setting a seed is not supported by `autotrain-advanced`.")
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+        Returns:
+          A list of predictions
+        """
+        self._logger.error("Use framework=`transformers` for inference.")
```
As we have the information to do so, I'd show here an example on how to load the trained model/s using `transformers`, just to serve as a tip

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+        Args:
+          output_dir (str): the path to save the model to
+        """
+        self._logger.warning("Models are saved on the Hugging Face hub, so this function is not supported.")
```
```suggestion
        self._logger.warning("Models are saved on the HuggingFace Hub, so this function is not supported.")
```

alvarobartt commented on src/argilla/training/base.py:
```diff
@@ -73,6 +73,9 @@ def __init__(
         self._name = name
+        if workspace is None:
+            workspace = rg.get_workspace()
+        self._workspace = workspace
```
```suggestion
        self._workspace = workspace or rg.get_workspace()
```

alvarobartt commented on src/argilla/training/autotrain_advanced.py:
```diff
@@ -0,0 +1,260 @@
+        return job_params
+
+
+class ArgillaAutoTrainTrainer(ArgillaTrainerSkeleton, AutoTrainer):
```
```suggestion
class ArgillaAutoTrainTrainer(ArgillaTrainerSkeleton, AutoTrainMixIn):
```

alvarobartt reviewed (commented):
Some minor comments, but great work overall! 👍🏻 Ideally, those suggestions should be applied in batch, but check those carefully as I haven't tested those myself

frascuchon committed:
feat: enable markdown configuration for text fields and questions (#3000)

# Description

This PR adds a param `use_markdown` to the `TextFieldSettings` and
`TextQuestionSettings` classes in API to enable markdown support. By
default, the parameter is disabled, which is compatible with the
previous behavior.


Also, some tests creating dataset fields and questions have been grouped
using the `pytest.mark.parameterize` decorator, since the internal
behavior was the same.

Refs #2998 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [x] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

New tests have been created 


**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: gabrielmbmb <gmartinbdev@gmail.com>

davidberenstein1957 committed:
suggestion batch Alvaro

chore: renamed mixin
chore: renamed variable name
chore: remove emojis from logging

Co-authored-by: Alvaro Bartolome <alvaro@argilla.io>

davidberenstein1957 committed:
docs: updated changelog
chore: added autotrain to environment_dev
chore: updated test naming
refac: started prepare

davidberenstein1957 committed:
chore: added clearer error for prediction mis-alignment

davidberenstein1957 committed:
Merge branch 'develop' into feat/autotrain-cherrypicked

alvarobartt commented on src/argilla/training/base.py:
```diff
@@ -261,6 +292,8 @@ def save(self, output_dir: str):
     def __init__(
         self,
+        name: str,
+        workspace: str,
```
I guess we can make this optional and default to `rg.get_workspace()`?

```suggestion
        workspace: Optional[str] = rg.get_workspace(),
```

Or the following instead:

```suggestion
        workspace: Optional[str] = None,
```

And then

```python
if workspace is None:
    workspace = rg.get_workspace()
...
```

WDYT? @davidberenstein1957

alvarobartt commented on tests/training/test_autotrain.py:
```diff
@@ -0,0 +1,77 @@
+
+FRAMEWORK = "autotrain"
+MODELS = ["prajjwal1/bert-tiny", "autotrain"]
+_HF_HUB_ACCESS_TOKEN = os.environ.get("HF_AUTH_TOKEN") or os.environ.get("HF_HUB_ACCESS_TOKEN")
```
Here indeed we could mock it as I did for the `push_to_huggingface` and `from_huggingface` methods in `FeedbackDataset`, but we can have a look at this later together 😄

davidberenstein1957 replied:
I agree!

alvarobartt reviewed (commented):
LGTM!

alvarobartt reviewed (commented):
LGTM!

davidberenstein1957 committed:
Apply suggestions from code review

Co-authored-by: Alvaro Bartolome <alvaro@argilla.io>

davidberenstein1957 committed:
Update environment_dev.yml

davidberenstein1957 committed:
fix: resolved un-renderable code

davidberenstein1957 committed:
Merge branch 'releases/1.8.0' into feat/autotrain-cherrypicked
